globus_action_provider_tools/testing/fixtures.py

```diff
@@ -0,0 +1,42 @@
+from unittest.mock import patch
+
+import pytest
+
+from globus_action_provider_tools.flask.apt_blueprint import ActionProviderBlueprint
+from globus_action_provider_tools.testing.mocks import mock_authstate, mock_tokenchecker
+
+
+@pytest.fixture(scope="module")
+def apt_blueprint_noauth():
+    """
+    A fixture designed to mock an ActionProviderBlueprint instance's Globus
+    Auth integration. The fixture returns a function to which the instance
+    should be supplied as a parameter:
+
+    i.e. apt_blueprint_noauth(aptb)
+    """
+
+    def _apt_blueprint_noauth(aptb: ActionProviderBlueprint):
+        # Manually remove the function that creates the internal token_checker
+        for f in aptb.deferred_functions:
+            if f.__name__ == "_create_token_checker":
+                aptb.deferred_functions.remove(f)
+
+        # Use a mocked token checker internally
+        aptb.checker = mock_tokenchecker()
+
+    return _apt_blueprint_noauth
+
+
+@pytest.fixture(scope="module")
+def flask_helpers_noauth():
+    """
+    A fixture designed to mock the Globus Auth integration in an Flask app
+    created using the api_helpers. Simply using this fixture will allow creating
+    the mocked app.
+    """
+    with patch(
```

Comment:
So, I see that my suggestion around monkeypatch might have repercussions down in the internal implementation. Maybe take a quick look to see if it does it in a nice way that keeps us consistent in the pytest world since we're already assuming it. If it makes it any bit harder, don't worry about it.

Reply:
Yea I think using `monkeypatch` internally would impose some limitations on how our libraries fixtures could be used (i.e. they could only be using in a certain scope). Looking at this again I've actually decided to broaden the scope of our fixtures to "session" (which is as I understand it the broadest scope possible) so that they can be used in fixtures of any scope. 

Using patches at this level maintains allows some extra flexibility for use cases, which may or may not be needed at this point.